### ISSUE TITLE
Ratatui.rs should have a nice website

### ISSUE THREAD

**Issue by joshka** [ratatui-org/ratatui-book#115]
It would be nice to have a landing page rather than ratatui.rs pointing directly at the book, so I slapped together a quick idea using [Zola](https://www.getzola.org/) with the [juice](https://www.getzola.org/themes/juice/) theme. I like the layout / content in [Acton](https://www.acton-lang.org/) as an idea for what we might want to add to a site using the theme in the part after the scroll.

I think we should keep the site and the book in the same repo (perhaps we could rename this repo to website) and build them to a single gh-pages site, putting the book content under `/book`. We'd need to create a header / theme that looks good for both the book and the main site. Currently we have a bunch of themes for this installed on the book, perhaps we could simplify this down to just two (dark/light)?

In theory we could use zola or some other SSG for both the site and docs, but then we'd have to manually write and maintain the things that mdbook gives us for free, so I think it's worth doing only the site part with zola and keeping the book with mdbook.

I've pushed this to https://github.com/joshka/ratatui-book/tree/zola-site if anyone wants to have a play with making this happen.

My preferences for this (unless there's compelling reasons to choose differently):
- the overarching goal should be minimal effort, max result
- static site generator that can push to gh-pages or cloudflare pages, rather than running a service somewhere
- rust based static site generator (e.g. zola) rather than adding extra languages / frameworks
- Minimal Javascript - just what we need for things to work - i.e. no SPA shenanigans or needing node just to build the site
- Site and book should look like they are part of the same site even though they are generated by mdbook / zola
- I'm a fan of dark navy blues and hues related to #102040. Starting the theme from that perspective would be nice to have perhaps with an orange accent?
- I know at least one of the ratatui devs is color blind, so we should this into consideration

Tasks:
- [ ] Create a nice looking landing page
- [ ] Consider whether any of the book pages belong in the site rather than the book (e.g. showcase)
- [ ] Create a header that works for the book and ties in to the main site
- [ ] Create a cohesive theme that works well for site and book (Zola uses SASS for its CSS - perhaps continue that)
- [ ] Update the CI workflows to deploy both site and book

![image](https://github.com/ratatui-org/ratatui-book/assets/381361/79854f48-361f-4b5a-bd85-d673e377811d)

**Comment by kdheepak** [comment]
Do you want to make a PR to a branch here? I can make commits to that to work on some of the tasks. I agree with everything you said!

> Currently we have a bunch of themes for this installed on the book, perhaps we could simplify this down to just two (dark/light)?

I like giving people options. Maybe we should create a theme for `mdbook` such that it matches the root page and make it the default. People can switch to other themes if they like.

**Comment by joshka** [comment]
> Do you want to make a PR to a branch here? I can make commits to that to work on some of the tasks. I agree with everything you said!

#116 

> I like giving people options. Maybe we should create a theme for mdbook such that it matches the root page and make it the default. People can switch to other themes if they like.

Sounds good. I do however like the simplicity of the dark / light switcher right on the title bar in https://etrojs.dev/ which also shows how a fixed title bar can work with their docs.

![image](https://github.com/ratatui-org/ratatui-book/assets/381361/4089d6b2-cc8e-42e7-ada3-5f40d183f36e)

**Comment by orhun** [comment]
The preview looks good! Maybe we can use a lighter blue though.

**Comment by kdheepak** [comment]
btw you make the PR from a branch in this PR instead of your own, it'll show the preview link in a GitHub comment.

**Comment by tz629** [comment]
I noticed that the [logo](https://github.com/joshka/ratatui-book/blob/zola-site/site/static/logo.png) used in the website has curved edges in some places and square edges in others, when it should probably be like [this](https://blog.orhun.dev/ratatui-logo.png) (pixel art). I think the problem here is image processing, which you shouldn't do on pixel art.

**Comment by joshka** [comment]
Yeah, I was lazy looking to do something as an idea and so just lasso clicked the background and deleted :) Thanks for catching that - much better!

**Comment by joshka** [comment]
> btw you make the PR from a branch in this PR instead of your own, it'll show the preview link in a GitHub comment.

I don't think we preview Zola sites right now without some extra magic to do it.

**Comment by joshka** [comment]
We can however do that with netlify - https://ratatui.netlify.app/ (e.g. https://github.com/joshka/ratatui-book/pull/3)

**Comment by joshka** [comment]
I had a bit of a play with Zola earlier. it's simple enough for a barebones SSG, but if we really want something nice we might want to put a bit more effort into things than just grabbing a template like this. I'm fine admitting my frontend web design skills are a little on the dated side (and there are too many frameworks to choose from) and so I'd defer to someone that has more skill in this area rather than continuing.

The main problem I saw making the book and the site seem like coherent whole was that mdbook and zola use different templating engines which means that it will be double the effort to make things look consistent.

So given that - I'd say there's a few options that would be worth considering:
1. Aim for two tools, mdbook + ??? (don't be fixated on Zola as an option if it causes us annoyance)
2. Aim to find a good tool that does docs + site together and switch from mdbook to that
3. Go ahead with Zola + mdbook

Either way - how about we move this issue to the main repo to see if we can find someone to help?

**Comment by kdheepak** [comment]
I'm fine with moving this issue to the main repo to discuss further.
